gh-95484: Improve `tempfile` docs for conditions when files are removed (#95485)
python/cpython: Doc/library/tempfile.rst

.. function:: TemporaryFile(mode='w+b', buffering=-1, encoding=None, newline=None, suffix=None, prefix=None, dir=None, *, errors=None)

   Return a :term:`file-like object` that can be used as a temporary storage area.

>    The file is created securely, using the same rules as :func:`mkstemp`.
>    Under Unix, the directory entry for the file is either not created at all or is
>    removed immediately after the file is created.  Other platforms do not support
>    this; your code should not rely on a temporary file created using this
>    function having or not having a visible name in the file system.
> 
>    The temporary file will be removed from the file system under the following conditions:
> 
>    - It is closed.
>    - or, it is garbage collected (and is implicitly closed)
>    - or, at the completion of the context if used as a context manager (see
>      :ref:`tempfile-examples`)

Reviewer: ```suggestion
   The file is created securely, using the same rules as :func:`mkstemp`,
   and the object can be used as a context manager
   (see :ref:`tempfile-examples`).

   Under Unix, the directory entry for the file is either not created at all or is
   removed immediately after the file is created.  Other platforms do not support
   this; your code should not rely on a temporary file created using this
   function having or not having a visible name in the file system.

   The file will be removed from the file system when the object is closed or
   garbage collected (and implicitly closed). Exiting the context manager
   closes the file.
```

Reviewer 2: I like it!

.. function:: TemporaryFile(mode='w+b', buffering=-1, encoding=None, newline=None, suffix=None, prefix=None, dir=None, *, errors=None)

   Return a :term:`file-like object` that can be used as a temporary storage area.
   The file is created securely, using the same rules as :func:`mkstemp`.
   Under Unix, the directory entry for the file is either not created at all or is
   removed immediately after the file is created.  Other platforms do not support
   this; your code should not rely on a temporary file created using this
   function having or not having a visible name in the file system.

>    The temporary file will be removed from the filesystem when the returned

Reviewer 2: This sentence is not quite grammatical. I'd suggest starting with "The temporary file will be removed from the file system under the following conditions:"
